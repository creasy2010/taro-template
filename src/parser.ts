import { join } from "path";
import { ILayoutNode, IParseConfig, IParseResult } from "./typings";
import { processStyle } from './style-processors';
import { generateCode } from './code-generators';


const dealImg = (config: IParseConfig, data: ILayoutNode) => {
  if (!config.imgDir) {
    config.imgDir = "./src/assets/image/";
  }
  config.imgDir = join(config.pwd, config.imgDir) + config.pagePath;
  config.fsExtra.ensureDirSync(config.imgDir);

  // 递归处理图片方法
  const dealImage = (data: ILayoutNode) => {
    // 如果当前结点是图片，处理当前结点
    if (data.type === "Image") {
      const imgSrc = downloadImg(data.attrs.src);
      data.attrs.src = imgSrc;
      data.attrs.source = imgSrc;
    }
    // 递归处理子节点
    data.children && data.children.forEach(i => {
      dealImage(i);
    });
  };

  // 下载图片方法
  let imgIdx = 0;
  const downloadImg = (src) => {
    let imgName = `${config.pageName}${imgIdx++}`;
    config.urllib.request(src, (err, data) => {
      config.fsExtra.writeFile(`${config.imgDir}/${imgName}.png`, data);
    });
    return imgName;
  };

  // 1.递归处理图片结点并下载图片
  dealImage(data);

}

/**
 * 获取模块的页面布局json
 * @param moduleId
 */
const getLayoutJson = async (config: IParseConfig): Promise<ILayoutNode> => {
  const url = `https://imgcook.taobao.org/api/getModule?moduleId=${JSON.parse(config.moduleId)}`;
  let { data, originData } = await new Promise((resolve) => {
    config.urllib.request(url, (err, data) => {
      try {
        const moduleData = JSON.parse(data.toString()).data;
        resolve({
          data: JSON.parse(moduleData.json),
          originData: JSON.parse(moduleData.originjson)
        });
      } catch (e) {
        throw new Error("读取imgcook layout数据失败");
      }
    });
  });

  const rmLineHeightPx = (style) => {
    let lineHeight = style.lineHeight;
    if (lineHeight && lineHeight.endsWith('px')) {
      style.lineHeight = parseInt(lineHeight.slice(0, lineHeight.length - 2));
    }
  }

  // 添加parent、设置默认值
  const addParent = (node: ILayoutNode) => {
    const { display, flexDirection, justifyContent, alignItems } = node.style;
    if (display === 'flex' && !flexDirection) node.style.flexDirection = 'row';
    if (display === 'flex' && !justifyContent) node.style.justifyContent = 'flex-start';
    if (display === 'flex' && !alignItems) node.style.alignItems = 'stretch';
    rmLineHeightPx(node.style);
    node.children.forEach(child => {
      child.parent = node;
      addParent(child);
    });
  };
  addParent(data);

  return data;
}



/**
 * 布局数据转页面代码
 * @param params
 */
export default async (config: IParseConfig): Promise<IParseResult> => {

  // 1.获取依赖数、预处理
  let data = await getLayoutJson(config);

  // 2.样式优化
  processStyle(data);

  // 3.图片处理
  dealImg(config, data);

  // 4.代码输出
  return generateCode(config, data);
};
