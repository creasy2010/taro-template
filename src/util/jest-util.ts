/**
 * @desc
 *
 * @使用场景
 *
 * @company qianmi.com
 * @Date    2019/7/25
 **/

import * as klaw from 'klaw';
import debug from  'debug';
const log  = debug('jest-util');
import {readFile} from 'fs-extra';
export interface IResult {
  [path: string]: string;
}

export function readDirFiles(dirPath: string): Promise<IResult> {
  log(`读取文件夹${dirPath}下文件内容`);
  dirPath =  dirPath.endsWith("/")?dirPath:dirPath+"/";
  return new Promise((resolve, reject) => {
    let allFilePaths: string[] = [];
    klaw(dirPath)
      .on('data', item => {
        if (item.stats.isFile()) {
          allFilePaths.push(item.path);
        }
      })
      .on('end', async () => {
        let result = {};
        let contents = await Promise.all(
          allFilePaths.map(filePath => readFile(filePath)),
        );
        for (let i = 0, iLen = allFilePaths.length; i < iLen; i++) {
          let fileName = allFilePaths[i].replace(dirPath, '');
          result[fileName] = contents[i].toString();
        }
        resolve(result);
      });
  });
}

export function sleep(time:number){
  return new Promise((resolve)=>{
    setTimeout(resolve,time);
  })
}