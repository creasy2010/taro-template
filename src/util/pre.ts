import { ILayoutNode } from "../typings";

/**
 * @desc
 *
 * @使用场景
 *
 * @coder.yang2010@gmail.com
 * @Date    2019/10/9
 **/



export function preClean (node: ILayoutNode):ILayoutNode{
  addParent(node);
  return node;
}



//去除px的单位;
const rmLineHeightPx = (style) => {
  let lineHeight = style.lineHeight;
  if (lineHeight && lineHeight.endsWith('px')) {
    style.lineHeight = parseInt(lineHeight.slice(0, lineHeight.length - 2));
  }
}

// 添加parent、设置默认值
const addParent = (node: ILayoutNode) => {
  const { display, flexDirection, justifyContent, alignItems } = node.style;
  if (display === 'flex' && !flexDirection) node.style.flexDirection = 'row';
  if (display === 'flex' && !justifyContent) node.style.justifyContent = 'flex-start';
  if (display === 'flex' && !alignItems) node.style.alignItems = 'stretch';
  rmLineHeightPx(node.style);
  node.children.forEach(child => {
    child.parent = node;
    addParent(child);
  });
};
