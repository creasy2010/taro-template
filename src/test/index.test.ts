import parser from '../parser';
import * as urllib from 'urllib';
import {join} from 'path';
import {tmpdir} from 'os';
import * as fsExtra from 'fs-extra';
import { readDirFiles, sleep } from "../util/jest-util";
/**
 * @desc
 *
 * @使用场景
 *
 * @coder.yang2010@gmail.com
 * @Date    2019/10/9
 **/

it('正常解析', async () => {
  let tmpDir =  join(tmpdir(), 'index.workbench')
  let config = {
    urllib,
    fsExtra,
    moduleId: '13398',
    pagePath: '/test',
    pageName: 'pageTitle',
    pwd: tmpDir,
  };

  let res = await parser(config);


  // console.log(res.mainComp.imports);
  fsExtra.ensureDirSync(`${join(config.pwd, `./data/${config.pageName}`)}`);

  await fsExtra.writeFile(
    `${join(config.pwd, `./data/${config.pageName}`)}/index.js`,
    res.mainComp.style,
  );
  await fsExtra.writeFile(
    `${join(config.pwd, `./data/${config.pageName}`)}/index.html`,
    res.mainComp.vdom,
  );
  await sleep(1000);
  let constent =  await readDirFiles(join(tmpDir,"data/pageTitle"));
  expect(constent).toMatchSnapshot('正常解析');
});
