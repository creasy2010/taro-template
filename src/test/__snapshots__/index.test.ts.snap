// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`正常解析: 正常解析 1`] = `
Object {
  "index.html": "<View style={styles.box}>
  <View style={styles.hd}>
    <Text style={styles.txt} allowFontScaling={false}>09:41</Text>
    <View style={styles.block}>
      <Image source={require('./img/pageTitle0.png')} style={styles.shebeiliuliang}/>
      <Image source={require('./img/pageTitle1.png')} style={styles.huiyuan}/>
      <View style={styles.group}>
        <View style={styles.colorDiv}>
        </View>
      </View>
      <Image source={require('./img/pageTitle2.png')} style={styles.dot}/>
    </View>
  </View>
  <View style={styles.bd}>
    <Image source={require('./img/pageTitle3.png')} style={styles.left}/>
    <Text style={styles.confirmOrder} allowFontScaling={false}>确认订单</Text>
    <View style={styles.block1}>
      <Image source={require('./img/pageTitle4.png')} style={styles.notice}/>
      <View style={styles.color}>
      </View>
      <Image source={require('./img/pageTitle5.png')} style={styles.renwuzhuangtaiDangqian}/>
    </View>
  </View>
  <View style={styles.submain}>
    <View style={styles.block2}>
      <View style={styles.block3}>
        <Text style={styles.timoma} allowFontScaling={false}>提莫妈</Text>
        <Text style={styles.text} allowFontScaling={false}>138****8138</Text>
      </View>
      <View style={styles.group1}>
        <Image source={require('./img/pageTitle6.png')} style={styles.div}/>
        <View style={styles.div}>
        </View>
      </View>
    </View>
    <Text style={styles.desc} allowFontScaling={false}>汪汪省龟龟市喵喵区鼠鼠路618号宠物大厦9-18</Text>
    <View style={styles.block4}>
      <View style={styles.colorDiv1}>
      </View>
      <View style={styles.color1}>
      </View>
      <View style={styles.colorDiv2}>
      </View>
      <View style={styles.color2}>
      </View>
      <View style={styles.colorDiv3}>
      </View>
      <View style={styles.color3}>
      </View>
      <View style={styles.colorDiv4}>
      </View>
      <View style={styles.color4}>
      </View>
      <View style={styles.colorDiv5}>
      </View>
      <View style={styles.color5}>
      </View>
      <View style={styles.colorDiv6}>
      </View>
      <View style={styles.color6}>
      </View>
      <View style={styles.colorDiv7}>
      </View>
      <View style={styles.color7}>
      </View>
      <View style={styles.colorDiv8}>
      </View>
      <View style={styles.color8}>
      </View>
    </View>
  </View>
  <View style={styles.row}>
    <Text style={styles.text} allowFontScaling={false}>配送方式</Text>
    <View style={styles.group2}>
      <Text style={styles.coupon} allowFontScaling={false}>快递配送</Text>
      <View style={styles.sWrap}>
        <Image source={require('./img/pageTitle7.png')} style={styles.s}/>
      </View>
    </View>
  </View>
  <View style={styles.row1}>
    <View style={styles.block5}>
      <Image source={require('./img/pageTitle8.png')} style={styles.img}/>
      <View style={styles.group3}>
        <View style={styles.group4}>
          <Text style={styles.summary} allowFontScaling={false}>大型犬只野兽派洗剪吹，凶悍大叔专业洗狗12年，洗遍天下无…</Text>
          <View style={styles.moneyWrap}>
            <Text style={styles.money} allowFontScaling={false}>¥198.9</Text>
            <View style={styles.wrap}>
              <Text style={styles.txt1} allowFontScaling={false}>×1</Text>
            </View>
          </View>
        </View>
        <Text style={styles.chicken} allowFontScaling={false}>鸡肉 400g</Text>
      </View>
    </View>
    <View style={styles.block5}>
      <Image source={require('./img/pageTitle9.png')} style={styles.img}/>
      <View style={styles.group5}>
        <View style={styles.priceWrap}>
          <Text style={styles.desc1} allowFontScaling={false}>大型犬野兽派洗剪吹A套餐</Text>
          <Text style={styles.price} allowFontScaling={false}>¥34</Text>
        </View>
        <Text style={styles.word} allowFontScaling={false}>×1</Text>
        <Text style={styles.chicken} allowFontScaling={false}>鸡肉 400g</Text>
      </View>
    </View>
    <View style={styles.block5}>
      <View style={styles.empty}>
      </View>
      <View style={styles.group5}>
        <View style={styles.group6}>
          <Text style={styles.summary1} allowFontScaling={false}>大型犬只野兽派洗剪吹，凶悍大叔专业洗狗12年，洗遍天下狗</Text>
          <View style={styles.moneyWrap}>
            <Text style={styles.money} allowFontScaling={false}>¥160.23</Text>
            <View style={styles.block6}>
              <Text style={styles.txt1} allowFontScaling={false}>×1</Text>
            </View>
          </View>
        </View>
        <Text style={styles.chicken} allowFontScaling={false}>鸡肉 400g</Text>
      </View>
    </View>
  </View>
  <View style={styles.main}>
    <Text style={styles.messageCanNotBeFilled} allowFontScaling={false}>留言 可不填</Text>
    <Text style={styles.desc2} allowFontScaling={false}>您可输入您的其他要求，最多50字</Text>
  </View>
  <View style={styles.row2}>
    <View style={styles.block7}>
      <Text style={styles.price1} allowFontScaling={false}>应付金额</Text>
      <Text style={styles.money1} allowFontScaling={false}>¥80.23</Text>
      <Text style={styles.price1} allowFontScaling={false}>商品金额</Text>
      <Text style={styles.price1} allowFontScaling={false}>¥23.24</Text>
    </View>
    <View style={styles.block8}>
      <Text style={styles.price2} allowFontScaling={false}>活动优惠</Text>
      <Text style={styles.price2} allowFontScaling={false}>-¥8.90</Text>
    </View>
    <View style={styles.block8}>
      <View style={styles.group7}>
        <Text style={styles.coupon} allowFontScaling={false}>优惠券</Text>
        <View style={styles.infoWrap}>
          <Text style={styles.info} allowFontScaling={false}>已选推荐优惠券</Text>
        </View>
      </View>
      <View style={styles.group8}>
        <Text style={styles.coupon} allowFontScaling={false}>-¥12.00</Text>
        <View style={styles.block9}>
          <Image source={require('./img/pageTitle10.png')} style={styles.div}/>
          <View style={styles.div}>
          </View>
        </View>
      </View>
    </View>
    <View style={styles.block10}>
      <Text style={styles.price2} allowFontScaling={false}>配送费用</Text>
      <Text style={styles.price2} allowFontScaling={false}>¥1.00</Text>
    </View>
  </View>
  <View style={styles.row}>
    <Text style={styles.text} allowFontScaling={false}>充值卡支付</Text>
    <View style={styles.priceWrap1}>
      <Text style={styles.price3} allowFontScaling={false}>￥100.00</Text>
      <View style={styles.block11}>
        <Image source={require('./img/pageTitle11.png')} style={styles.div}/>
        <View style={styles.div}>
        </View>
      </View>
    </View>
  </View>
  <View style={styles.ft}>
    <Text style={styles.price4} allowFontScaling={false}>还需支付 ¥1.00</Text>
    <View style={styles.block12}>
      <View style={styles.group9}>
        <View style={styles.weixinWrap}>
          <Image source={require('./img/pageTitle12.png')} style={styles.weixin}/>
        </View>
        <Text style={styles.text} allowFontScaling={false}>微信支付</Text>
      </View>
      <View style={styles.icongoumaichenggongWrap}>
        <Image source={require('./img/pageTitle13.png')} style={styles.div1}/>
        <View style={styles.div1}>
        </View>
      </View>
    </View>
  </View>
  <View style={styles.color9}>
  </View>
  <View style={styles.view}>
    <Text style={styles.text} allowFontScaling={false}>合计：¥198.23</Text>
    <View style={styles.tagWrap}>
      <Text style={styles.tag} allowFontScaling={false}>去支付</Text>
    </View>
  </View>
  <View style={styles.block13}>
    <View style={styles.colorDiv9}>
    </View>
  </View>
</View>",
  "index.js": "const styles = StyleSheet.create({
  box: {
    position: 'relative',
    backgroundColor: '#f5f5f5',
    height: 1080,
  },
  hd: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: '#ffffff',
    paddingRight: 14,
    paddingLeft: 17,
    height: 40,
  },
  txt: {
    color: '#000000',
    fontFamily: 'Helvetica',
    fontSize: 15,
  },
  block: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 15,
  },
  shebeiliuliang: {
    width: 17,
    height: 11,
    marginRight: 5,
  },
  huiyuan: {
    width: 15,
    height: 11,
    marginRight: 5,
  },
  group: {
    alignItems: 'flex-start',
    flexDirection: 'row',
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 2.666666746139526,
    borderColor: 'rgba(0,0,0,0.35)',
    paddingRight: 3,
    paddingLeft: 2,
    height: 11,
    marginRight: 1,
  },
  colorDiv: {
    marginTop: 1,
    borderRadius: 1.333333373069763,
    backgroundColor: '#000000',
    width: 18,
    height: 7,
  },
  dot: {
    width: 1,
    height: 4,
  },
  bd: {
    alignItems: 'center',
    flexDirection: 'row',
    backgroundColor: '#ffffff',
    height: 44,
    marginBottom: 4,
    paddingLeft: 8,
  },
  left: {
    width: 10,
    height: 18,
    marginRight: 132,
  },
  confirmOrder: {
    color: '#333333',
    fontSize: 18,
    marginRight: 53,
  },
  block1: {
    alignItems: 'center',
    flexDirection: 'row',
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 16,
    borderColor: 'rgba(235,235,235,1.00)',
    backgroundColor: '#ffffff',
    width: 88,
    height: 32,
    paddingLeft: 12,
    paddingRight: 12,
  },
  notice: {
    width: 19,
    height: 7,
    marginRight: 12,
  },
  color: {
    backgroundColor: '#cccccc',
    width: 1,
    height: 19,
    marginRight: 12,
  },
  renwuzhuangtaiDangqian: {
    width: 18,
    height: 18,
    marginRight: 12,
  },
  submain: {
    backgroundColor: '#ffffff',
    marginBottom: 12,
    paddingTop: 16,
  },
  block2: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 2,
    marginLeft: 12,
    marginRight: 11,
  },
  block3: {
    flexDirection: 'row',
  },
  timoma: {
    color: '#333333',
    fontSize: 16,
    marginRight: 8,
  },
  text: {
    color: '#333333',
    fontSize: 16,
  },
  group1: {
    position: 'relative',
    marginTop: 14,
    width: 4,
    height: 10,
  },
  div: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: 4,
    height: 10,
  },
  desc: {
    marginLeft: 12,
    color: '#b3b3b3',
    fontSize: 12,
    marginBottom: 16,
  },
  block4: {
    position: 'relative',
    alignItems: 'flex-start',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  colorDiv1: {
    position: 'absolute',
    top: 0,
    left: 0,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color1: {
    position: 'absolute',
    top: 0,
    right: 167,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv2: {
    position: 'absolute',
    top: 0,
    left: 96,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color2: {
    position: 'absolute',
    top: 0,
    right: 71,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv3: {
    position: 'absolute',
    top: 0,
    left: 48,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color3: {
    position: 'absolute',
    top: 0,
    right: 119,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv4: {
    position: 'absolute',
    top: 0,
    left: 144,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color4: {
    position: 'absolute',
    top: 0,
    right: 23,
    backgroundColor: '#fe7967',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv5: {
    position: 'absolute',
    top: 0,
    left: 24,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color5: {
    position: 'absolute',
    top: 0,
    right: 143,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv6: {
    position: 'absolute',
    top: 0,
    left: 120,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color6: {
    position: 'absolute',
    top: 0,
    right: 47,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv7: {
    position: 'absolute',
    top: 0,
    left: 72,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color7: {
    position: 'absolute',
    top: 0,
    right: 95,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  colorDiv8: {
    position: 'absolute',
    top: 0,
    left: 168,
    backgroundColor: '#7ca8ff',
    width: 16,
    height: 2,
    marginRight: 176,
  },
  color8: {
    backgroundColor: '#7ca8ff',
    width: 15,
    height: 2,
    marginRight: 176,
  },
  row: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: '#ffffff',
    paddingRight: 11,
    paddingLeft: 12,
    height: 48,
    marginBottom: 12,
  },
  group2: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  coupon: {
    color: '#333333',
    fontSize: 14,
    marginRight: 8,
  },
  sWrap: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    width: 20,
    height: 20,
  },
  s: {
    width: 13,
    height: 3,
  },
  row1: {
    backgroundColor: '#ffffff',
    paddingTop: 16,
    paddingBottom: 15,
    height: 296,
    marginBottom: 12,
  },
  block5: {
    flexDirection: 'row',
    marginBottom: 13,
    marginLeft: 12,
    marginRight: 12,
  },
  img: {
    borderRadius: 4,
    width: 80,
    height: 80,
    marginRight: 12,
  },
  group3: {
    height: 80,
    marginTop: 4,
  },
  group4: {
    flexDirection: 'row',
    marginBottom: 8,
  },
  summary: {
    color: '#333333',
    fontSize: 14,
    marginRight: 20,
  },
  moneyWrap: {
  },
  money: {
    color: '#333333',
    fontSize: 14,
    marginBottom: 8,
  },
  wrap: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginLeft: 13,
    marginRight: 13,
  },
  txt1: {
    color: '#b3b3b3',
    fontSize: 14,
  },
  chicken: {
    color: '#b3b3b3',
    fontSize: 12,
  },
  group5: {
    height: 80,
    marginTop: 4,
  },
  priceWrap: {
    flexDirection: 'row',
    marginBottom: 4,
  },
  desc1: {
    color: '#333333',
    fontSize: 14,
    marginRight: 69,
  },
  price: {
    color: '#333333',
    fontSize: 14,
  },
  word: {
    marginLeft: 244,
    color: '#b3b3b3',
    fontSize: 14,
    marginBottom: 8,
  },
  empty: {
    opacity: 1.00,
    borderRadius: 2,
    backgroundImage: 'linear-gradient(to right, rgb(247,247,247), rgb(245,245,245))',
    width: 80,
    height: 80,
    marginRight: 12,
  },
  group6: {
    flexDirection: 'row',
    marginBottom: 8,
  },
  summary1: {
    color: '#333333',
    fontSize: 14,
    marginRight: 11,
  },
  block6: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginLeft: 18,
    marginRight: 18,
  },
  main: {
    justifyContent: 'center',
    backgroundColor: '#ffffff',
    height: 80,
    marginBottom: 12,
  },
  messageCanNotBeFilled: {
    marginLeft: 12,
    color: '#333333',
    fontSize: 16,
    marginBottom: 18,
  },
  desc2: {
    marginLeft: 12,
    color: '#b3b3b3',
    fontSize: 16,
  },
  row2: {
    backgroundColor: '#ffffff',
    paddingTop: 16,
    paddingBottom: 15,
    height: 166,
    marginBottom: 12,
  },
  block7: {
    flexDirection: 'row',
    marginBottom: 14,
    flexWrap: 'wrap',
    marginLeft: 1,
  },
  price1: {
    color: '#333333',
    fontSize: 14,
    marginRight: 249,
  },
  money1: {
    color: '#ff6600',
    fontSize: 14,
    marginRight: 249,
  },
  block8: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 14,
    marginLeft: 12,
    marginRight: 11,
  },
  price2: {
    color: '#333333',
    fontSize: 14,
  },
  group7: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  infoWrap: {
    alignItems: 'center',
    flexDirection: 'row',
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: 'rgba(255,102,0,1.00)',
    backgroundColor: '#ffffff',
    paddingRight: 9,
    paddingLeft: 8,
    height: 20,
  },
  info: {
    color: '#ff6600',
    fontSize: 12,
  },
  group8: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 20,
  },
  block9: {
    position: 'relative',
    width: 4,
    height: 10,
  },
  block10: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 14,
    marginLeft: 12,
    marginRight: 11,
  },
  priceWrap1: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 16,
  },
  price3: {
    color: '#ff6600',
    fontSize: 14,
    marginRight: 8,
  },
  block11: {
    position: 'relative',
    width: 4,
    height: 10,
  },
  ft: {
    justifyContent: 'center',
    backgroundColor: '#ffffff',
    height: 104,
  },
  price4: {
    marginLeft: 12,
    color: '#333333',
    fontSize: 16,
    marginBottom: 16,
  },
  block12: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginLeft: 12,
    marginRight: 11,
  },
  group9: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  weixinWrap: {
    alignItems: 'center',
    flexDirection: 'row',
    borderRadius: 6,
    backgroundColor: '#3ecf78',
    paddingRight: 9,
    paddingLeft: 8,
    height: 40,
    marginRight: 8,
  },
  weixin: {
    width: 24,
    height: 20,
  },
  icongoumaichenggongWrap: {
    position: 'relative',
    marginTop: 10,
    width: 20,
    height: 20,
  },
  div1: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: 20,
    height: 20,
  },
  color9: {
    backgroundColor: '#ffffff',
    height: 94,
    boxShadow: 'inset 0px 0.5px 0px rgba(235, 235, 235, 1.00)',
    position: 'absolute',
    bottom: 0,
    left: 0,
  },
  view: {
    bottom: 34,
    left: 0,
    boxShadow: 'inset 0px 0.5px 0px rgba(235, 235, 235, 1.00)',
    position: 'absolute',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: '#ffffff',
    paddingRight: 11,
    paddingLeft: 12,
    height: 60,
  },
  tagWrap: {
    alignItems: 'center',
    flexDirection: 'row',
    borderRadius: 6,
    backgroundImage: 'linear-gradient(to top right, rgba(254, 154, 41, 1), rgba(254, 90, 20, 1))',
    paddingRight: 27,
    paddingLeft: 27,
    height: 40,
  },
  tag: {
    color: '#ffffff',
    fontSize: 14,
  },
  block13: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    alignItems: 'flex-start',
    flexDirection: 'row',
    justifyContent: 'center',
    height: 34,
  },
  colorDiv9: {
    marginTop: 22,
    borderRadius: 2,
    backgroundColor: '#000000',
    width: 134,
    height: 4,
  },
});",
}
`;
